client/packages/invoices/src/Prescriptions/api/hooks/usePrescriptionLines.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  DeletePrescriptionLineInput,
+  InsertPrescriptionLineInput,
+  InvoiceLineNodeType,
+  RecordPatch,
+  UpdatePrescriptionLineInput,
+  useMutation,
+} from '@openmsupply-client/common';
+import { usePrescription } from './usePrescription';
+import { DraftStockOutLine } from 'packages/invoices/src/types';
+import { usePrescriptionGraphQL } from '../usePrescriptionGraphQL';
+import { PrescriptionRowFragment } from '../operations.generated';
+import { PRESCRIPTION, PRESCRIPTION_LINE } from './keys';
+import { createInputObject, mapStatus } from './utils';
+
+// Hook to manage prescription lines. Only has "save" and "delete"
+// functionality, as the query is done as part of the full prescription query
+// (usePrescription).
+//
+// We don't manage draft state in here, as there is a lot of complex logic
+// associated with it, which is handled by the `useDraftPrescriptionLines` hook.
+
+export const usePrescriptionLines = (id?: string) => {
+  const {
+    query: { data },
+  } = usePrescription(id);
+
+  // SAVE LINES
+  const {
+    mutateAsync: updateMutation,
+    isLoading: isSavingLines,
+    error: saveLineError,
+  } = useSaveLines(data?.id ?? '', data?.invoiceNumber ?? -1);
+
+  const saveLines = async ({
+    draftPrescriptionLines,
+    patch,
+  }: {
+    draftPrescriptionLines: DraftStockOutLine[];
+    patch?: RecordPatch<PrescriptionRowFragment>;
+  }) => {
+    await updateMutation({
+      draftPrescriptionLines,
+      patch,
+    });
+  };
+
+  // DELETE LINES
+  const {
+    mutateAsync: deleteMutation,
+    isLoading: isDeletingLines,
+    error: deleteLinesError,
+  } = useDeleteLines(data?.invoiceNumber ?? -1);
+
+  const deleteLines = async (rowsToDelete: DraftStockOutLine[]) => {
+    const lines = rowsToDelete.map(({ id }) => ({ id }));
+    await deleteMutation(lines);
+  };
+
+  return {
+    save: { saveLines, isSavingLines, saveLineError },
+    delete: { deleteLines, isDeletingLines, deleteLinesError },
+  };
+};
+
+const useSaveLines = (id: string, invoiceNum: number) => {
+  const { prescriptionApi, storeId, queryClient } = usePrescriptionGraphQL();
+
+  const mutationFn = async ({
+    draftPrescriptionLines,
+    patch,
+  }: {
+    draftPrescriptionLines: DraftStockOutLine[];
+    patch?: RecordPatch<PrescriptionRowFragment>;
+  }) => {
+    if (patch && id !== '') patch.id = id;
+    const input = {
+      insertPrescriptionLines: draftPrescriptionLines
+        .filter(
+          ({ type, isCreated, numberOfPacks }) =>
+            isCreated &&
+            type === InvoiceLineNodeType.StockOut &&
```

Comment:
Any reason why they wouldn't be the right type here?

Reply:
Not that I can think of, but this is just copied over from the original. Worth removing you think?